### PR TITLE
doc(centroid): document each impl and include example

### PR DESCRIPTION
**Pull request by RobWalt**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Hey everyone!

This PR just includes documentation and some examples of the implementations of the `Centroid` trait for all the `geo-types`.

I also moved some existing docs to the impl of the `centroid` function and I moved the impl of the trait for `Triangle` further up below the impl of the trait for `Rect` since I think this grouping makes more sense.

### CODE REVIEW

File: geo/src/algorithm/centroid.rs [hunk 1]
```diff
@@ -63,6 +63,24 @@ where
 {
     type Output = Point<T>;
 
+    /// The Centroid of a [`Line`] is it's middle point
```
Comment (urschrei) [inline geo/src/algorithm/centroid.rs:66]: Should be "its"!

Comment (RobWalt) [reply geo/src/algorithm/centroid.rs:66]: Fixing this up everywhere in the file, had multiple occurences

**Review by frewsxcv — APPROVED** [review]
thank you sooo much! these docs will be so helpful to users 🎉

**Pull request merged**